docs/maintainers/Release_Process.md

```diff
@@ -1,48 +1,64 @@
 < [Documentation](../Readme.md) / Release Process
 
-## Branches
+# Branches
 Our branching strategy is very similar to GitFlow
-* `master` branch has release commits. One commit - one release. Each release commit should be tagged with version `maj.min.patch`
-* `beta` branch is for stabilization of ongoing release. We don't merge new features into `beta` branch - only bug fixes are acceptable. Hotfixes for already released software should be delivered to this branch as well.
-* `develop` branch is a main branch for ongoing development. Pull requests with new features should be targeted to this branch, `develop` version is one step ahead of 'beta`
+- `master` branch has release commits. One commit - one release. Each release commit should be tagged with version `maj.min.patch`
+- `beta` branch is for stabilization of ongoing release. Only safe changes should be targeted into this branch. Breaking changes (e.g. save format changes) are forbidden in beta.
```

Comment:
In the classic waterfall model, there is feature complete milestone, which means that no new features are supposed to be introduced. Then, sqa can focus on testing for those new features. Making new feature as last minute change is bad pattern because leaves less time for testing => poor quality in the end.
Definition of "safe" is quite unclear also but the main question is why do we have to rush and cannot deliver features into develop with longer stabilization time and more time to fix the regression issues.

Reply:
Is there any need to postpone small, finished features that have minimal chance of regressions AND are frequently asked by players till next release in 3 months?

In the same time, bugfixes for some tricky issues can be actually more dangerous than such minor features. 

Detection of misclick on touch input is technically a feature, but amount of "pain" it causes our players is quite high, so postponing it for 3 months is likely worse option since some controls are barely usable at the moment.

Czech translation is clearly a feature, but chance of regressions is basically zero, so downsides of including it into a hotfix are minimal.

Another problem is that line between "bugfix" and "new feature" can be blurry sometimes.
Should we count "give 500 XP on town capture" as feature or as bug? This can be viewed as bug since this is OH3 behavior. Or as feature since VCMI never had handling of such case.

Reply:
Agree, that bug definition is not very strict since we don't have requirements, but I would say, there are two types of bugs can be (but not must) fixed during stabilization period:
 - something which worked earlier but doesn't work now (regression)
 - newly introduced feature doesn't work as expected (low-quality new functionality)
there are probably one more category, which is game mechanic issue (e.g. behavior is contradicting to requirements, h3 has a quite good game mechanic documentation). Example with 500XP fits this category

The rest are expceptions. And if there are ready features frequently asked by players, why not just simply shift release date?
The main point is do not consider "beta" as "develop" and do not hollow scope and dedicate efforts to improve quality of features we are delivering.

Reply:
impact on gameplay matters as well, for example there is artifact backpack in 1.3.2 that is basically isolated functionality that will not break rest of game, and is one of top wanted HD mod features (also it was almost done for 1.3.0)

Reply:
> And if there are ready features frequently asked by players, why not just simply shift release date?

What do you mean? Release 1.4.0 with just these "frequently asked features" like artifact backpack? This is definitely not a major release like our 1.X.0 releases are.

And releasing 1.4.0 "as it" sooner means another save game format breakage, which is already a problem on some platforms.

> The main point is do not consider "beta" as "develop"

But we do not? A lot of risky features went in develop and not it beta.

I also don't remember any regressions introduced during stabilization.

Reply:
no, I meant delay 1.3.0 if there is very desired feature pending, especially taking into consideration "also it was almost done for 1.3.0"

We are talking about documentation, so my suggestion here is only about elaboration for "safe" meaning.